### PR TITLE
Javascript load

### PR DESCRIPTION
**Pull request by aloupfor**

### CODE REVIEW

File: en/developer_guide/block_display.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
+Block display
+=============
+
+Display Independence
+--------------------
+
+In Front Office, some blocks need specific stylesheets and javascript file.
+To perform this, the main node layout loads the requireJS library which allows each block to call require() function.
```
Comment (ngilain) [inline en/developer_guide/block_display.rst]: to call the ``require()`` function of the RequireJS library

File: en/developer_guide/block_display.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+To allow stylesheets dynamic load, Open Orchestra offers an AMD (Asynchronous Module Definition) named openOrchestraCSS
```
Comment (ngilain) [inline en/developer_guide/block_display.rst]: an AMD module

File: en/developer_guide/block_display.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+.. code-block:: javascript
+
+    requirejs(
+        ['jquery'],
+        function() {
+            require(
+                [
+                    //Define required elements
+                    'openOrchestraCss', //Module used to load stylesheets
+                    ''/bundles/nameofbundle/block/NameOfMyBlock/js/foo.js'', //Specific js in block
```
Comment (ngilain) [inline en/developer_guide/block_display.rst]: too many ' here :)

File: en/developer_guide/block_display.rst [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                    //You can add optional javascript code here
+                    jQuery(document).ready(function ($) {
+
+
+                    });
+                }
+            )
+        }
+    );
+
```
Comment (ngilain) [inline en/developer_guide/block_display.rst:54]: You have said all that was to say, but I wonder if the explanation would be in the code block as comment or outside.

File: en/developer_guide/block_display.rst [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+Example: Gallery block
+----------------------
+
+In MediaBundle, Resources/views/Block directory contains the Front Media blocks display templates.
```
Comment (alavieille) [inline en/developer_guide/block_display.rst]: `Resources/views/Block`, use ``

File: en/developer_guide/block_display.rst [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+The file Gallery/show.html.twig ends with the blockLoader.js load which does stylesheet and javascript dynamic load
```
Comment (alavieille) [inline en/developer_guide/block_display.rst]: `Gallery/show.html.twig` : use ``

File: en/developer_guide/block_display.rst [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+in relation with Gallery block.
+
+.. code-block::
```
Comment (alavieille) [inline en/developer_guide/block_display.rst]: use a javascript code block : 

.. code-block:: javascript

**Comment by alavieille** [comment]
May be add a little explanation of requireJS and a link to her documentation

File: en/developer_guide/block_display.rst [hunk 8]
```diff
@@ -0,0 +1,95 @@
+Block display
+=============
+
+Display Independence
+--------------------
+
+In Front Office, some blocks need specific stylesheets and javascript file.
```
Comment (nicolasThal) [inline en/developer_guide/block_display.rst]: In the Front Office, some blocks could require some specific ...

File: en/developer_guide/block_display.rst [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+To perform this, the main node layout loads the requireJS library which is a JavaScript file and module loader
```
Comment (nicolasThal) [inline en/developer_guide/block_display.rst]: To allow any block to add some specific file, the main ...

Comment (nicolasThal) [inline en/developer_guide/block_display.rst]: the `requireJs` library

you should put `` around it

File: en/developer_guide/block_display.rst [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+(see `requireJS <http://requirejs.org/>`_).
```
Comment (nicolasThal) [inline en/developer_guide/block_display.rst]: the link to the requireJs page should be on the first time you give the techno name

File: en/developer_guide/block_display.rst [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+Each block is allowed to call the ``require()`` function of the requireJS library.
+To allow stylesheets dynamic load, Open Orchestra offers an AMD module named openOrchestraCSS which is meant to be use
```
Comment (nicolasThal) [inline en/developer_guide/block_display.rst]: could you split the line with less characters on it

File: en/developer_guide/block_display.rst [hunk 12]
```diff
@@ -0,0 +1,103 @@
+Block display
+=============
+
+Display Independence
+--------------------
+
+In the Front Office, some blocks could require some specific stylesheets and javascript file.
```
Comment (alavieille) [inline en/developer_guide/block_display.rst]: files

File: en/developer_guide/block_display.rst [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+----------------------
+
+In MediaBundle, ``Resources/views/Block`` directory contains the Front Media blocks display
+templates.
+The file ``Gallery/show.html.twig`` ends with the ``blockLoader.js`` load which does stylesheet
+and javascript dynamic load in relation with Gallery block.
+
+.. code-block:: javascript
+
+    src=require(['{{ asset('/bundles/openorchestramedia/block/Gallery/blockLoader.js') }}']);
```
Comment (alavieille) [inline en/developer_guide/block_display.rst]: `<script>require(['{{ asset('/bundles/openorchestramedia/block/Gallery/blockLoader.js') }}']);</script>
`

**Comment by aloupfor** [comment]
@itkg-alavieille @itkg-ngilain update done

File: en/developer_guide/block_display.rst [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+Each block is allowed to call the ``require()`` function of the ``requireJS`` library.
+To allow stylesheets dynamic load, Open Orchestra offers an AMD module named ``openOrchestraCSS``
+which is meant to be use with ``requireJS``.
+In a case in which the same block is displayed several times, stylesheets and javascript files
+corresponding will be loaded only once.
+
+Block loading file
+-------------------
+
+For each block there is a ``blockLoader.js`` file which use the ``requireJS`` library to load the
```
Comment (ngilain) [inline en/developer_guide/block_display.rst]: file using the `require.js`

**Comment by ngilain** [comment]
:+1: with the little modification

File: en/developer_guide/block_display.rst [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+
+Display Independence
+--------------------
+
+In the Front Office, some blocks could require some specific stylesheets and javascript files.
+To allow any block to add some specific files, the main node layout loads the
+`requireJS`_ library which is a JavaScript file and module loader.
+Each block is allowed to call the ``require()`` function of the ``requireJS`` library.
+To allow stylesheets dynamic load, Open Orchestra offers an AMD module named ``openOrchestraCSS``
+which is meant to be use with ``requireJS``.
```
Comment (juchi) [inline en/developer_guide/block_display.rst]: to be used

File: en/developer_guide/block_display.rst [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+In a case in which the same block is displayed several times, stylesheets and javascript files
+corresponding will be loaded only once.
```
Comment (juchi) [inline en/developer_guide/block_display.rst]: corresponding stylesheets and JavaScript files will only be loaded once

File: en/developer_guide/block_display.rst [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+
+Block loading file
+-------------------
+
+For each block there is a ``blockLoader.js`` file using the ``requireJS`` library to load the
+requirements needed to initialize a block.
```
Comment (juchi) [inline en/developer_guide/block_display.rst]: "to load the requirements" => it's unclear what "requirements" are, are they resource files ?

File: en/developer_guide/block_display.rst [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+The  required elements are the module used to load stylesheets: ``openOrchestraCss``,
+specific javascript files in block and javascript files in common libraries.
```
Comment (juchi) [inline en/developer_guide/block_display.rst]: i guess this answers my previous remark, but it should come before, maybe the 2 paragraphs should be rephrased

File: en/developer_guide/block_display.rst [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+
+After loading the elements, it does the initialization. During this step, javascript is executed
+and stylesheets are loaded using ``openOrchestraCss`` module with ``load()`` function.
```
Comment (juchi) [inline en/developer_guide/block_display.rst]: the ... module with the ... function

File: en/developer_guide/block_display.rst [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+
+Block loading file
+-------------------
+
+For each block there is a ``blockLoader.js`` file using the ``requireJS`` library to load the
+requirements needed to initialize a block.
+The  required elements are the module used to load stylesheets: ``openOrchestraCss``,
+specific javascript files in block and javascript files in common libraries.
+
+After loading the elements, it does the initialization. During this step, javascript is executed
```
Comment (juchi) [inline en/developer_guide/block_display.rst]: "the JavaScript code is run" whould be better
But, it should be precised what code is run, is it only the code specific to a block ?

File: en/developer_guide/block_display.rst [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+                }
+            )
+        }
+    );
+
+
+Example: Gallery block
+----------------------
+
+In MediaBundle, ``Resources/views/Block`` directory contains the Front Media blocks display
```
Comment (juchi) [inline en/developer_guide/block_display.rst]: In the MediaBundle, the ... directory

File: en/developer_guide/block_display.rst [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+
+    <script>
+        require(['{{ asset('/bundles/openorchestramedia/block/Gallery/blockLoader.js') }}']);
+    </script>
+
+
+Blocs maintainability
+---------------------
+
+To keep consistency between all blocks, you should use a file tree as the following:
```
Comment (juchi) [inline en/developer_guide/block_display.rst]: like the following

**Comment by juchi** [comment]
General remarks : 
I think "stylesheet" is longer to read than "CSS", and it's used several times here
JavaScript should be written with the capital J and S

**Comment by aloupfor** [comment]
@juchi update done

**Comment by juchi** [comment]
I don't know if it's normal, but you have written openOrchestraCSS and then openOrchestraCss
good for me otherwise :+1:

**Pull request merged**